From: Kalesh AP <kalesh.purayil@emulex.com>
Date: Thu, 17 Jul 2014 16:20:21 +0530
Subject: be2net: return -ETIMEDOUT when a FW-cmd times out
Patch-mainline: v3.17-rc1
Git-commit: fd45160cef0a4782cde70ec4cdeb9421ea8460b6
References: bnc#908322 FATE#317535

When the FW stops responding with completions, return -ETIMEDOUT error
(instead of -1) to the stack.
Signed-off-by: Kalesh AP <kalesh.purayil@emulex.com>
Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/emulex/benet/be_cmds.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/net/ethernet/emulex/benet/be_cmds.c
+++ b/drivers/net/ethernet/emulex/benet/be_cmds.c
@@ -2222,7 +2222,7 @@ int lancer_cmd_write_object(struct be_ad
 
 	if (!wait_for_completion_timeout(&adapter->et_cmd_compl,
 					 msecs_to_jiffies(60000)))
-		status = -1;
+		status = -ETIMEDOUT;
 	else
 		status = adapter->flash_status;
 
@@ -2318,7 +2318,7 @@ int be_cmd_write_flashrom(struct be_adap
 
 	if (!wait_for_completion_timeout(&adapter->et_cmd_compl,
 					 msecs_to_jiffies(40000)))
-		status = -1;
+		status = -ETIMEDOUT;
 	else
 		status = adapter->flash_status;
 
